evalml/pipelines/components/utils.py

```diff
@@ -130,7 +130,6 @@ def fit(self, X, y):
         Returns:
             self
         """
-        X, y = check_X_y(X, y)
```

Comment:
Removing [this](https://scikit-learn.org/stable/modules/generated/sklearn.utils.check_X_y.html). Useful, but not necessary if we're not trying to be fully scikit-learn compliant, and allows us to pass pandas data structures to fit instead of numpy (which causes weird / incorrect type-inference).

Reply:
Should we also make this change in the regressor?

Reply:
Yep, same question :)

Reply:
Yup, gonna remove that!